app/presenters/flow_presenter.rb

```diff
@@ -79,8 +79,15 @@ def presenter_for(node)
   end
 
   def response_for_current_question
-    responses = params[:responses]
-    responses[current_state.current_node.to_s]
+    if response_store
+      responses = params[:responses]
+      responses[current_state.current_node.to_s]
+    elsif params[:previous_response].present?
+      params[:previous_response]
+    else
+      question_number = current_state.path.size
+      all_responses[question_number]
+    end
```

Comment:
I feel like object/responsiblity wise it'd make a bit more sense if the current response was part of the current_state of a smart answer, there are some things feel a bit clunky around this. As we already discussed this is something I'm going to try look into later on and isn't blocking this PR (it'd be intended to enhance it). 

I do like that this method encapsulates the clunkiness though and that all the views only need to reference response_for_current_question.